core/src/navigation/ProductSwitcher.html

```diff
@@ -0,0 +1,83 @@
+<div class="fd-shellbar__action fd-shellbar__action--collapsible">
+  <div class="fd-product-switcher">
+    <div class="fd-popover fd-popover--right">
+      <div class="fd-popover__control" on:click="event.stopPropagation()">
+        <button
+          class="fd-button--shell sap-icon--grid"
+          aria-expanded="{dropDownStates.productSwitcherPopover || false}"
+          aria-haspopup="true"
+          on:click="toggleDropdownState('productSwitcherPopover')"
+        ></button>
+        <div
+          class="fd-popover__body fd-popover__body--right"
+          aria-hidden="{!(dropDownStates.productSwitcherPopover || false)}"
+          id="productSwitcherPopover"
+        >
+          <div class="fd-product-switcher__body">
+            <nav>
+              <ul>
+                {#if productSwitcherItems && productSwitcherItems.length} {#each
+                productSwitcherItems as productSwitcherItem}
+                <li on:click="onActionClick(productSwitcherItem)">
+                  <span class="fd-product-switcher__product-icon"
+                    ><img src="{productSwitcherItem.icon}"
+                  /></span>
+                  <span class="fd-product-switcher__product-title"
+                    >{productSwitcherItem.label}</span
+                  >
+                </li>
+                {/each} {/if}
+              </ul>
+            </nav>
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+</div>
+
+<script type="text/javascript">
+  import { LuigiConfig } from '../services/config.js';
+  import * as Routing from '../services/routing.js';
+  export default {
+    async oncreate() {
+      const productSwitcherConfig = LuigiConfig.getConfigValue(
+        'navigation.productSwitcher'
+      );
+      this.set({
+        config: productSwitcherConfig
+      });
+
+      if (productSwitcherConfig) {
+        const productSwitcherItems = await LuigiConfig.getConfigValueAsync(
+          'navigation.productSwitcher.items'
+        );
+        this.set({
+          productSwitcherItems
+        });
+      }
+    },
+    methods: {
+      onActionClick(productSwitcherItem) {
+        if (productSwitcherItem.externalLink) {
```

Comment:
Please wrap this `if ... else` under the unsaved changes modal promise. You can take the same implementation from Context Switcher component:
```
goTo(path) {
        this.root.getUnsavedChangesModalPromise().then(() => {
          Routing.navigateTo(path);
        });
      }, 
```
This is not reproducible with the links you added, since there is a fallback mechanism in case you don't wrap it under the unsaved changes modal promise, but this is messing the browser history a little bit. 
You can reproduce the issue if you have unsaved changes (can be activated in the overview page) and use in the product switcher for example a link to an external page which opens in the same tab. Currently it is navigating to that page, but it should display the unsaved changes modal instead.

Reply:
done

Reply:
This is still not working for external links opening in the same tab. My proposal would be:
```
    methods: {
      onActionClick(productSwitcherItem) {
        this.root.getUnsavedChangesModalPromise().then(() => {
          if (productSwitcherItem.externalLink) {
            window.open(
              productSwitcherItem.externalLink.url,
              productSwitcherItem.externalLink.sameWindow ? '_self' : '_blank'
            );
          } else {
            Routing.navigateTo(productSwitcherItem.link);
          }
        });
      },
      toggleDropdownState(name) {
        this.fire('toggleDropdownState', { name });
        const dropDownStates = this.get().dropDownStates || {};
      }
    }
```